src/rpc/quorums.cpp

```diff
@@ -789,6 +789,59 @@ static UniValue quorum_rotationinfo(const JSONRPCRequest& request, const LLMQCon
     return quorumRotationInfoRet.ToJson();
 }
 
+static void quorum_dkginfo_help(const JSONRPCRequest& request)
+{
+    RPCHelpMan{"quorum dkginfo",
+               "Return information regarding DKGs.\n"
+               "Enabled only for Masternode and works only when SPORK_17_QUORUM_DKG_ENABLED spork is ON.\n",
+               {
+                       {},
+               },
+               RPCResults{},
+               RPCExamples{""},
+    }.Check(request);
+}
+
+static UniValue quorum_dkginfo(const JSONRPCRequest& request, const LLMQContext& llmq_ctx, const ChainstateManager& chainman)
+{
+    quorum_dkginfo_help(request);
+
+    if (!fMasternodeMode) {
+        throw JSONRPCError(RPC_INVALID_REQUEST, "RPC allowed only for Masternodes");
+    }
+
+    llmq::CDKGDebugStatus status;
+    llmq_ctx.dkg_debugman->GetLocalDebugStatus(status);
+    UniValue ret(UniValue::VOBJ);
+    ret.pushKV("nActiveDKGs", int(status.sessions.size()));
+
+    if (status.sessions.empty()) {
+        int nTipHeight = WITH_LOCK(cs_main, return chainman.ActiveChain().Height());
+        const Consensus::Params &consensus_params = Params().GetConsensus();
+
+        auto minNextDKG = [&consensus_params, nTipHeight]() {
+            int minDkgWindow = std::numeric_limits<int>::max();
+            for (const auto &params: consensus_params.llmqs) {
+                if (!params.useRotation)
+                    minDkgWindow = std::min(minDkgWindow, params.dkgInterval - (nTipHeight % params.dkgInterval));
+                else {
+                    if (nTipHeight % params.dkgInterval > params.signingActiveQuorumCount) {
+                        // Next potential DKG is the DKG for quorumIndex 0
+                        minDkgWindow = std::min(minDkgWindow, params.dkgInterval - (nTipHeight % params.dkgInterval));
+                    }
+                        // We are currently during a rotation cycle. Since sessions.empty(), next we return next potential DKG is 1 (next block for next quorumIndex)
+                    else {
+                        minDkgWindow = std::min(minDkgWindow, 1);
+                    }
+                }
+            }
+            return minDkgWindow;
+        };
+        ret.pushKV("nextDKG", minNextDKG());
```

Comment:
Something like this would be better
```suggestion
        int minDkgWindow = std::numeric_limits<int>::max();
        for (const auto &params: consensus_params.llmqs) {
            minDkgWindow = std::min(minDkgWindow, [&](){
                if (!params.useRotation) return params.dkgInterval - (nTipHeight % params.dkgInterval);
                if (nTipHeight % params.dkgInterval > params.signingActiveQuorumCount) {
                    // Next potential DKG is the DKG for quorumIndex 0
                    return params.dkgInterval - (nTipHeight % params.dkgInterval);
                }
                // We are currently during a rotation cycle. Since sessions.empty(), next we return next potential DKG is 1 (next block for next quorumIndex)
                return 1;
            }());
        };
        ret.pushKV("nextDKG", minDkgWindow);
```

probably even even better
```suggestion
        int minDkgWindow = std::transform_reduce(
            consensus_params.llmqs.begin(), 
            consensus_params.llmqs.end(),
            std::numeric_limits<int>::max(), // Initial value for the reduction
            [](int a, int b) { return std::min(a, b); }, // Reducing function
            [nTipHeight](const auto &params) { // Transformation
                if (!params.useRotation) 
                    return params.dkgInterval - (nTipHeight % params.dkgInterval);
                
                if (nTipHeight % params.dkgInterval > params.signingActiveQuorumCount)
                    return params.dkgInterval - (nTipHeight % params.dkgInterval);
                
                // We are currently during a rotation cycle. Since sessions.empty(),
                // next we return next potential DKG is 1 (next block for next quorumIndex)
                return 1;
            }
        );
        
        ret.pushKV("nextDKG", minDkgWindow);
```

Reply:
I think I prefer the first suggestion.

Reply:
Sadly I think the second one is "more correct" shows intent a bit clearer maybe, and may be optimized easier 🤷 not a super strong opinion on either

Reply:
Don't be sad! @knst @UdjinM6 opinions ?

Reply:
![image](https://github.com/dashpay/dash/assets/545784/7c0eebe2-ac7b-4e79-b2d9-337227b70d30)

Anyway, 
this part is confusing:
```
                        minDkgWindow = std::min(minDkgWindow, 1);
```
can it be zero? Can it be less than 1? Can it be negative? That question that I asked myself when I read code.

also I have several nits. I would write this code like:
```
     if (status.sessions.empty()) {
        const int nTipHeight{WITH_LOCK(cs_main, return chainman.ActiveChain().Height())};

        auto minNextDKG = [](Consensus::Params& consensusParams, int nTipHeight) {
            int minDkgWindow{std::numeric_limits<int>::max()};
            for (const auto& params: consensus_params.llmqs) {
                if (params.useRotation && (nTipHeight % params.dkgInterval <= params.signingActiveQuorumCount)) {
                    return 1;
                }
                minDkgWindow = std::min(minDkgWindow, params.dkgInterval - (nTipHeight % params.dkgInterval));
            }
            return minDkgWindow;
        };
        ret.pushKV("nextDKG", minNextDKG(Params().GetConsensus(), nTipHeight));
    }
```

Reply:
Decided to go with @knst suggestion. It looks more readable I think